.github/workflows/dart.yml

```diff
@@ -0,0 +1,95 @@
+name: Dart CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dart-lang/setup-dart@v1
+      - name: Dart version
+        run: |
+          dart --version
+          uname -a
+      - name: Install dependencies
+        run: dart pub get
+      - name: Upgrade dependencies
+        run: dart pub upgrade
```

Comment:
In theory I agree with you, until I had some real world issues.

I only add the upgrade command because in some cases the GitHub workflow has come cached packages and is not getting the last version for some pre-installed packages.

Dart pub get only gets the last version if it's downloading the package for the first time.

Reply:
`dart pub get`

`If the system cache doesn’t already contain the dependencies, dart pub get updates the cache, downloading dependencies if necessary.`

From:
https://dart.dev/tools/pub/cmd/pub-get

When running `dart pub get`, if a dependency is already in the cache and matches the constraint, it won't download a newer version.

Reply:
That isn't my experience with how it works, but I am fine with using `dart pub upgrade`. We definitely don't need both though (just remove the `get` then).